public/app/features/explore/ContentOutline/ContentOutline.tsx

```diff
@@ -56,62 +98,106 @@ export function ContentOutline({ scroller, panelId }: { scroller: HTMLElement |
 
     reportInteraction('explore_toolbar_contentoutline_clicked', {
       item: 'select_section',
-      type: buttonTitle,
+      type: itemPanelId,
     });
   };
 
   const toggle = () => {
-    toggleExpanded();
+    toggleContentOutlineExpanded();
     reportInteraction('explore_toolbar_contentoutline_clicked', {
       item: 'outline',
-      type: expanded ? 'minimize' : 'expand',
+      type: contentOutlineExpanded ? 'minimize' : 'expand',
     });
   };
 
+  const toggleSection = () => {
+    toggleSectionExpanded();
+    reportInteraction('explore_toolbar_contentoutline_clicked', {
+      item: 'section',
+      type: sectionExpanded ? 'minimize' : 'expand',
+    });
+  };
+
+  const outlineItemsHaveChildren = outlineItems.some((item) => item.children);
+
   useEffect(() => {
-    const activeItem = outlineItems.find((item) => {
-      const top = item?.ref?.getBoundingClientRect().top;
+    let activeItem = null;
 
-      if (!top) {
-        return false;
+    // Loop through each item
+    for (const item of outlineItems) {
+      let top = item?.ref?.getBoundingClientRect().top;
+
+      // Check item
+      if (top && top >= 0) {
+        activeItem = item;
+        break;
       }
 
-      return top >= 0;
-    });
+      // Check children
+      const activeChild = item.children?.find((child) => {
+        let childTop = child?.ref?.getBoundingClientRect().top;
+        return childTop && childTop >= QUERY_ROW_TOP_OFFSET;
+      });
+
+      if (activeChild) {
+        activeItem = activeChild;
```

Comment:
Could we have multiple activeItems (or "active" as part of the state of a single item) to have parent and child highlighted at the same time?

Reply:
Yes, that's my plan moving forward

Reply:
@ifrost this is working now